dynamic_programming/shortest_common_supersequence.cpp

```diff
@@ -0,0 +1,185 @@
+
+            // making supersequence
+            // i and j are initially pointed towards end of strings
+            // Super-sequence will be constructed backwards
+            int i=n1;
+            int j=n2;
+            std::string s;
+      
+            while(i>0 && j>0) {
+
+                // If the characters at i and j of both strings are same
+                // We only need to add them once in s
+                if(str1[i-1] == str2[j-1]) {
+                    s.push_back(str1[i-1]);
+                    i--;
+                    j--;
+                }
+                // otherwise we check lookup table for recurrences of characters
+                else {
+                    if(lookup[i-1][j] > lookup[i][j-1]) {
+                        s.push_back(str1[i-1]);
+                        i--;
+                    }
+                    else {
+                        s.push_back(str2[j-1]);
+                        j--;
+                    }
+                }
+            }
+
+            // copying remaining elements
+            // if j becomes 0 before i
+            while(i > 0) {
+                s.push_back(str1[i-1]);
+                i--;
+            }
+
+            // if i becomes 0 before j
+            while(j > 0) {
+                s.push_back(str2[j-1]);
+                j--;
+            }
+
+            // As the super sequence is constructd backwards
+            // reversing the string before returning gives us the correct output  
+            reverse(s.begin(), s.end());
+            return s;
+        }
+    } // namespace shortest_common_supersequence
+} // namespace dynamic_programming
+
+/** 
+ * Test Function
+ * @return void 
+*/
+static void test() {
+
+    // custom input vector
+    std::vector <std::vector <std::string>> scsStrings {
+        {"ABCXYZ", "ABZ"},
+        {"ABZ", "ABCXYZ"},
+        {"AGGTAB", "GXTXAYB"},
+        {"X", "Y"},
+    };
+
+    // calculated output vector by scs function
+    std::vector <std::string> calculatedOutput;
+    for(auto & scsString : scsStrings) {
+        int n1 = scsString[0].length();
+        int n2 = scsString[1].length();
+        
+        calculatedOutput.push_back(dynamic_programming::shortest_common_supersequence::scs(
+            scsString[0], scsString[1], n1, n2
+        ));
+    }
+
+    // expected output vector acc to problem statement
+    std::vector <std::string> expectedOutput {
+        "ABCXYZ",
+        "ABCXYZ",
+        "AGGXTXAYB",
+        "XY"
+    };
+
+    // Testing implementation via assert function
+    // It will throw error if any of the expected test fails
+    // Else it will give nothing
+    for(int i=0; i < scsStrings.size(); i++) {
+        assert(expectedOutput[i] == calculatedOutput[i]);
+    }
+
+    std::cout << "All tests passed successfully!\n";
+    return;
+}
+
+/** Main function (driver code)*/
+int main() {
+
+    // test for implementation
+    std::cout << "\nTesting implementation\n";
+    test();
```

Comment:
```suggestion
    test();
    std::cout << "\nTest implementations passed!\n";
```

Reply:
No need for `std::cout << "\nTesting implementations passed!\n"` as this is already present in `test()` function